src/main/java/com/leverx/learningmanagementsystem/service/impl/CourseServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+package com.leverx.learningmanagementsystem.service.impl;
+
+import com.leverx.learningmanagementsystem.dto.course.CreateCourseDto;
+import com.leverx.learningmanagementsystem.entity.Course;
+import com.leverx.learningmanagementsystem.entity.CourseSettings;
+import com.leverx.learningmanagementsystem.entity.Lesson;
+import com.leverx.learningmanagementsystem.entity.Student;
+import com.leverx.learningmanagementsystem.exception.EntityValidationException.EntityNotFoundException;
+import com.leverx.learningmanagementsystem.exception.EntityValidationException.IncorrectResultSizeException;
+import com.leverx.learningmanagementsystem.mapper.course.CourseMapper;
+import com.leverx.learningmanagementsystem.repository.CourseRepository;
+import com.leverx.learningmanagementsystem.repository.CourseSettingsRepository;
+import com.leverx.learningmanagementsystem.repository.LessonRepository;
+import com.leverx.learningmanagementsystem.repository.StudentRepository;
+import com.leverx.learningmanagementsystem.service.CourseService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class CourseServiceImpl implements CourseService {
+
+    private final CourseRepository courseRepository;
+    private final LessonRepository lessonRepository;
+    private final StudentRepository studentRepository;
+    private final CourseSettingsRepository courseSettingsRepository;
+    private final CourseMapper courseMapper;
+
+    @Override
+    public Course getById(UUID id) {
+        log.info("Get course with id {}", id);
+        return courseRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Course with id = " + id + " not found"));
+    }
+
+    @Override
+    @Transactional
+    public List<Course> getAll() {
+        log.info("Get all courses");
+        return courseRepository.findAll();
+    }
+
+    @Override
+    @Transactional
+    public Course create(CreateCourseDto createCourseDto) {
```

Comment:
service layer should accept already prepared entity instead of DTO
the same for other methods